doc/howtos/rdtraining/02_setup.rst

```diff
@@ -0,0 +1,432 @@
+
+
+The following instructions are based on the official `GitHub documentation <https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh>`__.
+
+
+Here is a step-by-step procedure to do so:
+
+
+- Generate a new SSH key, add it to the ssh-agent and copy the SSH key to your clipboard.
+
+  .. code-block:: console
+
+    $ ssh-keygen -t ed25519 -C "xyz@odoo.com"
+    $ ssh-add ~/.ssh/id_ed25519
+    $ sudo apt-get install xclip
+    $ xclip -sel clip < ~/.ssh/id_ed25519.pub
+
+
+On Github:
+
+
+- In the upper-right corner of any page, click your profile photo, then click Settings
+
+  .. image:: 02_setup/media/userbar-account-settings.png
+
+- In the user settings sidebar, click SSH and GPG keys.
+
+  .. image:: 02_setup/media/settings-sidebar-ssh-keys.png
+
+- Click New SSH key or Add SSH key.
+
+  .. image:: 02_setup/media/ssh-add-ssh-key.png
+
+- In the "Title" field, add a descriptive label for the new key.
+- Paste your key into the "Key" field.
+
+  .. image:: 02_setup/media/ssh-key-paste.png
+
+- Click Add SSH key.
+
+
+Fetch the sources
+-----------------
+
+All the Odoo sources will be located in `$HOME/src/`
+
+.. code-block:: console
+
+    $ mkdir -p $HOME/src
+    $ cd $HOME/src
+    $ git clone git@github.com:odoo/odoo.git
+    $ git clone git@github.com:odoo/enterprise.git
+
+.. tip:: Cloning the repositories will take a while, enjoy a cup of coffee while waiting.
+
+.. tip:: You may need to ask your manager read rights to fetch the enterprise repository.
+
+.. _howto/rdtraining/02_setup/development_repository:
+
+Configure development repository
+--------------------------------
+
+To contribute to Odoo's development, you will need to
+`fork the repository <https://guides.github.com/activities/forking/>`__, create a branch containing
+your code in the fork and submit a
+`Pull Request <https://docs.github.com/en/github/getting-started-with-github/github-glossary#pull-request>`__
+to the Odoo repository.
+
+In the case you are lucky enough to work at Odoo, the forks already exist. They are called
+`odoo-dev/odoo` and `odoo-dev/enterprise`.
+
+.. code-block:: console
+
+    $ cd  $HOME/src/odoo
+    $ git remote add odoo-dev git@github.com:odoo-dev/odoo.git #add odoo-dev as a new remote
+    $ git remote rename origin odoo #change the name of origin (the odoo repository) to odoo
+    $ git remote set-url --push odoo no_push #remove the possibility to push to odoo (you can only push to odoo-dev)
+
+    $ cd  $HOME/src/enterprise
+    $ git remote add enterprise-dev git@github.com:odoo-dev/enterprise.git
+    $ git remote rename origin enterprise
+    $ git remote set-url --push enterprise no_push
+
+
+Useful git commands
+-------------------
+
+These are useful git commands for your day-to-day work.
+
+* Change branch:
+  When you change branches, both repositories (odoo and enterprise) must be synchronized, i.e. both
```

Comment:
I agree, although I think it's a bit out-of-scope at this point. Indeed, the trainee is not supposed to switch branches during the training.

I also agree it's not a hard requirement, but in practice it will lead to errors. Actually, if we wanted to be completed, we should mention that both repositories should be on the same branch, but both branches should have been fetched within a short period of time. Having `odoo` at day `D` and `enterprise` at day `D-x` can already lead to errors.